src/main/java/com/autotune/metrics/KruizeNotificationCollectionRegistry.java

```diff
@@ -0,0 +1,90 @@
+                HashMap<Integer, RecommendationNotification> termLevelNotificationHashMap = termRecommendations.getNotifications();
+                createCounterTag("term", termName, null, termLevelNotificationHashMap.values());
+                if (null != termRecommendations.getRecommendationForModelHashMap()) {
+                    for (Map.Entry<String, MappedRecommendationForModel> recommendationForModel : termRecommendations.getRecommendationForModelHashMap().entrySet()) {
+                        String modelName = recommendationForModel.getKey();
+                        MappedRecommendationForModel mappedRecommendationForModel = recommendationForModel.getValue();
+                        HashMap<Integer, RecommendationNotification> modelNotificationHashMap = mappedRecommendationForModel.getNotificationHashMap();
+                        createCounterTag("model", termName, modelName, modelNotificationHashMap.values());
+                    }
+                }
+            }
+        }
+    }
+
+    /**
+     * Creates a counter with tags for the given level, term, model, and list of recommendation notifications.
+     *
+     * @param level                          The level of the notification (e.g., container, timestamp, term, model).
+     * @param term                           The term associated with the notification.
+     * @param model                          The cost or performance model associated with the notification.
+     * @param recommendationNotificationList The list of recommendation notifications to create counters for.
+     */
+    public void createCounterTag(String level, String term, String model, Collection<RecommendationNotification> recommendationNotificationList) {
+        for (RecommendationNotification recommendationNotification : recommendationNotificationList) {
+            Tags additionalTags = Tags.empty();
+            if (("|" + KruizeDeploymentInfo.log_recommendation_metrics_level + "|").contains("|" + recommendationNotification.getType() + "|") == true) {
+                additionalTags = additionalTags.and(KruizeConstants.KRUIZE_RECOMMENDATION_METRICS.TAG_NAME, String.format(KruizeConstants.KRUIZE_RECOMMENDATION_METRICS.notification_format, this.experiment_name, this.container_name, KruizeConstants.DateFormats.simpleDateFormatForUTC.format(this.interval_end_time), level, term, model, String.valueOf(recommendationNotification.getCode()), recommendationNotification.getType(), recommendationNotification.getMessage()));
+                Counter counterNotifications = MetricsConfig.meterRegistry().find(KruizeConstants.KRUIZE_RECOMMENDATION_METRICS.METRIC_NAME).tags(additionalTags).counter();
+                if (counterNotifications == null) {
+                    counterNotifications = MetricsConfig.timerBKruizeNotifications.tags(additionalTags).register(MetricsConfig.meterRegistry);
+                }
+                counterNotifications.increment();
```

Comment:
What happens if generateRecommendations is called twice for the same time interval and experiment ?

Reply:
Counter gets incremented